plugins/woocommerce/i18n/units.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Units
+ *
+ * Returns a multidimensional array of measurement units and their labels.
+ * Unit labels should be defined in English and translated native through localization files.
+ *
+ * @package WooCommerce\i18n
+ * @version
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+return array(
+	'weight'     => array(
+		'kg'  => _x( 'kg', 'weight unit kilograms', 'woocommerce' ),
```

Comment:
There may not be an easy answer to this one, and really I'm just being mindful of the fact that some recent and otherwise valid i18n changes have inadvertently caused some measure of disruption: by switching from non-contextual to contextual strings, I think we'd effectively be throwing out the [previous translations](https://github.com/woocommerce/woocommerce/pull/36708/files#diff-7c6852728aa4a52b0f1420a2b6a886a902fb32230da64ca8415b9ab752cad32aL115). 

Is there a way to minimize the disruption that might cause for our merchants (not sure if there *is* a technical solution, but perhaps we could issue a CTA for community translators, via the dev blog)?

Reply:
> I think we'd effectively be throwing out the [previous translations](https://github.com/woocommerce/woocommerce/pull/36708/files#diff-7c6852728aa4a52b0f1420a2b6a886a902fb32230da64ca8415b9ab752cad32aL115).

From [this](https://make.wordpress.org/polyglots/handbook/translating/glotpress-translate-wordpress-org/#strings-status-color-key), I think you're right. The strings would probably be changed to "fuzzy" status, so they'd have to be reapproved before they'd go back into the language pack.

Adding the context seems like the right thing to do here from a translator UX perspective, since something like "g" out of context doesn't mean much to them. But this might be better done in conjunction with other improvements around continuous deployment, where new and changed translations would get submitted to Polyglots right away instead of only when a new version is released.

I'll change it back to the simple `__()` function for now.

Reply:
Changed back in 9691159

Note that this necessitates a slight change in the code snippet used for testing, updated above.